private/buf/cmd/buf/command/generate/generate_test.go

```diff
@@ -1036,22 +1012,21 @@ plugins:
 				fullOutputPath,
 			)
 			require.NoError(t, err)
-			if useCleanFlag || fullOutputPath == expectedPluginConfigCleanOut {
-				require.True(t, len(data) > 1, "expected non-fake data at %q", fullOutputPath)
-			} else {
-				require.True(t, len(data) == 1, "expected fake data at %q", fullOutputPath)
-			}
+			// Always expect non-fake data, because the existing ".jar" or ".zip"
+			// file is always replaced by the output. This is the existing and correct
+			// behavior.
+			require.True(t, len(data) > 1, "expected non-fake data at %q", fullOutputPath)
```

Comment:
Flagging this. It seems the clean option doesn't affect `.jar` and `.zip` outputs since they have always been cleaned (the generator rewrites the `.jar` or `.zip` file), prior to the `--clean` flag and `clean` option.